### ISSUE TITLE
The performance is much lower than your trained model when training and deeplabv3+ on Pascal Context

### ISSUE THREAD

**Issue by Magic-Ha** [open-mmlab/mmsegmentation#394]
I get a much worse performance when training deeplabv3+ on Pascal Context val, using 4*2080 ti  4imgs/GPU
I ran the training code by:
```
CUDA_VISIBLE_DEVICES=0,1,2,3 ./tools/dist_train.sh configs/deeplabv3plus/deeplabv3plus_r101-d8_480x480_40k_pascal_context.py 4 --launcher pytorch --work-dir work_dirs/Deeplab/pc/test/
```
which has a huge difference from your resnet101 iter40000 deeplabv3+ pascal context training log

So I want to know your training setting on Pascal Context 
or could you tell me how to train the model using 4*2080 ti( now I'm just using the default setting, and my config file is down below:

'''
norm_cfg = dict(type='SyncBN', requires_grad=True)
model = dict(
    type='EncoderDecoder',
    pretrained='open-mmlab://resnet101_v1c',
    backbone=dict(
        type='ResNetV1c',
        depth=101,
        num_stages=4,
        out_indices=(0, 1, 2, 3),
        dilations=(1, 1, 2, 4),
        strides=(1, 2, 1, 1),
        norm_cfg=dict(type='SyncBN', requires_grad=True),
        norm_eval=False,
        style='pytorch',
        contract_dilation=True),
    decode_head=dict(
        type='DepthwiseSeparableASPPHead',
        in_channels=2048,
        in_index=3,
        channels=512,
        dilations=(1, 12, 24, 36),
        c1_in_channels=256,
        c1_channels=48,
        dropout_ratio=0.1,
        num_classes=60,
        norm_cfg=dict(type='SyncBN', requires_grad=True),
        align_corners=False,
        loss_decode=dict(
            type='CrossEntropyLoss', use_sigmoid=False, loss_weight=1.0)),
    auxiliary_head=dict(
        type='FCNHead',
        in_channels=1024,
        in_index=2,
        channels=256,
        num_convs=1,
        concat_input=False,
        dropout_ratio=0.1,
        num_classes=60,
        norm_cfg=dict(type='SyncBN', requires_grad=True),
        align_corners=False,
        loss_decode=dict(
            type='CrossEntropyLoss', use_sigmoid=False, loss_weight=0.4)),
    train_cfg=dict(),
    test_cfg=dict(mode='slide', crop_size=(480, 480), stride=(320, 320)))
dataset_type = 'PascalContextDataset'
data_root = 'data/VOCdevkit/VOC2010/'
img_norm_cfg = dict(
    mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True)
img_scale = (520, 520)
crop_size = (480, 480)
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations'),
    dict(type='Resize', img_scale=(520, 520), ratio_range=(0.5, 2.0)),
    dict(type='RandomCrop', crop_size=(480, 480), cat_max_ratio=0.75),
    dict(type='RandomFlip', prob=0.5),
    dict(type='PhotoMetricDistortion'),
    dict(
        type='Normalize',
        mean=[123.675, 116.28, 103.53],
        std=[58.395, 57.12, 57.375],
        to_rgb=True),
    dict(type='Pad', size=(480, 480), pad_val=0, seg_pad_val=255),
    dict(type='DefaultFormatBundle'),
    dict(type='Collect', keys=['img', 'gt_semantic_seg'])
]
test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(
        type='MultiScaleFlipAug',
        img_scale=(520, 520),
        flip=False,
        transforms=[
            dict(type='Resize', keep_ratio=True),
            dict(type='RandomFlip'),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_rgb=True),
            dict(type='ImageToTensor', keys=['img']),
            dict(type='Collect', keys=['img'])
        ])
]
data = dict(
    samples_per_gpu=4,
    workers_per_gpu=4,
    train=dict(
        type='PascalContextDataset',
        data_root='data/VOCdevkit/VOC2010/',
        img_dir='JPEGImages',
        ann_dir='SegmentationClassContext',
        split='ImageSets/SegmentationContext/train.txt',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(type='LoadAnnotations'),
            dict(type='Resize', img_scale=(520, 520), ratio_range=(0.5, 2.0)),
            dict(type='RandomCrop', crop_size=(480, 480), cat_max_ratio=0.75),
            dict(type='RandomFlip', prob=0.5),
            dict(type='PhotoMetricDistortion'),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_rgb=True),
            dict(type='Pad', size=(480, 480), pad_val=0, seg_pad_val=255),
            dict(type='DefaultFormatBundle'),
            dict(type='Collect', keys=['img', 'gt_semantic_seg'])
        ]),
    val=dict(
        type='PascalContextDataset',
        data_root='data/VOCdevkit/VOC2010/',
        img_dir='JPEGImages',
        ann_dir='SegmentationClassContext',
        split='ImageSets/SegmentationContext/val.txt',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(
                type='MultiScaleFlipAug',
                img_scale=(520, 520),
                flip=False,
                transforms=[
                    dict(type='Resize', keep_ratio=True),
                    dict(type='RandomFlip'),
                    dict(
                        type='Normalize',
                        mean=[123.675, 116.28, 103.53],
                        std=[58.395, 57.12, 57.375],
                        to_rgb=True),
                    dict(type='ImageToTensor', keys=['img']),
                    dict(type='Collect', keys=['img'])
                ])
        ]),
    test=dict(
        type='PascalContextDataset',
        data_root='data/VOCdevkit/VOC2010/',
        img_dir='JPEGImages',
        ann_dir='SegmentationClassContext',
        split='ImageSets/SegmentationContext/val.txt',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(
                type='MultiScaleFlipAug',
                img_scale=(520, 520),
                flip=False,
                transforms=[
                    dict(type='Resize', keep_ratio=True),
                    dict(type='RandomFlip'),
                    dict(
                        type='Normalize',
                        mean=[123.675, 116.28, 103.53],
                        std=[58.395, 57.12, 57.375],
                        to_rgb=True),
                    dict(type='ImageToTensor', keys=['img']),
                    dict(type='Collect', keys=['img'])
                ])
        ]))
log_config = dict(
    interval=50, hooks=[dict(type='TextLoggerHook', by_epoch=False)])
dist_params = dict(backend='nccl')
log_level = 'INFO'
load_from = None
resume_from = None
workflow = [('train', 1)]
cudnn_benchmark = True
optimizer = dict(type='SGD', lr=0.004, momentum=0.9, weight_decay=0.0001)
optimizer_config = dict()
lr_config = dict(policy='poly', power=0.9, min_lr=0.0001, by_epoch=False)
runner = dict(type='IterBasedRunner', max_iters=40000)
checkpoint_config = dict(by_epoch=False, interval=4000)
evaluation = dict(interval=4000, metric='mIoU')
work_dir = 'work_dirs/Deeplab/pc/test/'
gpu_ids = range(0, 1)

'''

**Comment by xvjiarui** [comment]
Hi @Magic-Ha 
Have you tried test provided checkpoint?

**Comment by Magic-Ha** [comment]
> Hi @Magic-Ha
> Have you tried test provided checkpoint from our model zoo?

yes, I tested the 40k resnet 101 deeplabv3+ checkpoint that you provided with no aug-test, but only got 
'''
+--------+-------+-------+-------+
| Scope  | mIoU  | mAcc  | aAcc  |
+--------+-------+-------+-------+
| global | 47.23 | 56.85 | 74.25 |
+--------+-------+-------+-------+
'''
Is that because there is something wrong with my pascal context dataset?
cause after I download and converted the dataset as you docs said, I only got 5104 val imgs.

**Comment by xvjiarui** [comment]
It is almost the same as the model zoo. Which number are you comparing?

**Comment by Jun-jieChen** [comment]
@Magic-Ha 
I meet the same problem, I also got 4996 train images and 5104 val images. I don't know the difference between `Pascal Context` and `Pascal Context 59`.

**Comment by Magic-Ha** [comment]
> @Magic-Ha
> I meet the same problem, I also got 4996 train images and 5104 val images. I don't know the difference between `Pascal Context` and `Pascal Context 59`.

It seems that Pascal Context 59 just ignore the background during evaluation， it has the same number of images as ‘ Pascal Context’.

**Comment by Jun-jieChen** [comment]
> > @Magic-Ha
> > I meet the same problem, I also got 4996 train images and 5104 val images. I don't know the difference between `Pascal Context` and `Pascal Context 59`.
> 
> It seems that Pascal Context 59 just ignore the background during evaluation， it has the same number of images as ‘ Pascal Context’.

But, if you look at the `log` files, the `Pascal Context` has 257 epochs, however, the `Pascal Context 59` only has 129 epochs with 40k iterations.

**Comment by Magic-Ha** [comment]
> > > @Magic-Ha
> > > I meet the same problem, I also got 4996 train images and 5104 val images. I don't know the difference between `Pascal Context` and `Pascal Context 59`.
> > 
> > 
> > It seems that Pascal Context 59 just ignore the background during evaluation， it has the same number of images as ‘ Pascal Context’.
> 
> But, if you look at the `log` files, the `Pascal Context` has 257 epochs, however, the `Pascal Context 59` only has 129 epochs with 40k iterations.

Oh, I did not notice that mmsegmentation has updated the context 59. I just ignore the background during training and inference using almost the same config file as Pascal Context back then, and achieved 53.4% mIoU with deeplabV3+.

**Comment by Jun-jieChen** [comment]
> > > > @Magic-Ha
> > > > I meet the same problem, I also got 4996 train images and 5104 val images. I don't know the difference between `Pascal Context` and `Pascal Context 59`.
> > > 
> > > 
> > > It seems that Pascal Context 59 just ignore the background during evaluation， it has the same number of images as ‘ Pascal Context’.
> > 
> > 
> > But, if you look at the `log` files, the `Pascal Context` has 257 epochs, however, the `Pascal Context 59` only has 129 epochs with 40k iterations.
> 
> Oh, I did not notice that mmsegmentation has updated the context 59. I just ignore the background during training and inference using almost the same config file as Pascal Context back then, and achieved 53.4% mIoU with deeplabV3+.

Wow! How can you do that? Could you give me some tips?

**Comment by Magic-Ha** [comment]
> > > > > @Magic-Ha
> > > > > I meet the same problem, I also got 4996 train images and 5104 val images. I don't know the difference between `Pascal Context` and `Pascal Context 59`.
> > > > 
> > > > 
> > > > It seems that Pascal Context 59 just ignore the background during evaluation， it has the same number of images as ‘ Pascal Context’.
> > > 
> > > 
> > > But, if you look at the `log` files, the `Pascal Context` has 257 epochs, however, the `Pascal Context 59` only has 129 epochs with 40k iterations.
> > 
> > 
> > Oh, I did not notice that mmsegmentation has updated the context 59. I just ignore the background during training and inference using almost the same config file as Pascal Context back then, and achieved 53.4% mIoU with deeplabV3+.
> 
> Wow! How can you do that? Could you give me some tips?

I just remove the background class in the class 'PascalContextDataset' ( located in './mmseg/datasets/pascal_context.py' ). I just ckecked the update, and found that they did the same (the only different between class 'PascalContextDataset' and 'PascalContextDataset59' is that the background class is removed in 'PascalContextDataset59'). 

However, my performance(53.4% ms+flip) is still lower than the just posted results(54.67% ms+flip), maybe I will train it again using the official config setting later for reproducting their results.

**Comment by Jun-jieChen** [comment]
@Magic-Ha 
Thanks for your reply and help!

**Comment by Magic-Ha** [comment]
> @Magic-Ha
> Thanks for your reply and help!

Just remembered! You should also reduce `0` (represent background class) in the original label. I think the official config and impletation is more clear than mine ! Maybe you should take a look at the config file `_base_/dataset/pascal_context_59.py`.

**Comment by Jun-jieChen** [comment]
@

> > @Magic-Ha
> > Thanks for your reply and help!
> 
> Just remembered! You should also reduce `0` (represent background class) in the original label. I think the official config and impletation is more clear than mine ! Maybe you should take a look at the config file `_base_/dataset/pascal_context_59.py` updated 6 days ago.

OK!

**Comment by xiexinch** [comment]
Closing the issue, as there is no activity for a while.
We hope your issue has been resolved. 
If not, please feel free to open a new one.
